Basic multiple files analysis

lukechurch opened the pull request:
@devoncarew PTAL

This adds the ability to analyse multiple dart files via a new EP 'analyzeMulti'. It doesn't link that through to the RPC APIs yet.

It cleans a bit more data betweens calls, but I don't think this will cause a measurable performance impact.

devoncarew commented on lib/src/analyzer.dart:
```diff
@@ -90,13 +105,65 @@ class Analyzer {
+      });
+
+      List<_Error> errors = errorInfos
+      .expand((AnalysisErrorInfo info) {
```
indent += 2

lukechurch replied:
done

devoncarew commented on lib/src/api_classes.dart:
```diff
@@ -55,6 +57,24 @@ class SourceRequest {
+
+  @ApiProperty(
+      description: 'Name of file that the offset is in')
+  String offsetName;
```
Having the `offset` and `offsetName` be separate fields seems a little awkward. `offset` could be a `String`, `int` tuple?

lukechurch replied:
I'll add a tiny 'Location' class to handle this.

devoncarew commented:
lgtm, some comments

lukechurch commented:
Thanks @devoncarew with simple tests I couldn't get a difference that was measurable. I'll try again with a larger test set, but it looks like it's going to be fine wrt performance.

I've addressed the review comments.

lukechurch commented:
I will add an additional benchmark for multi-files in a separate PR.

lukechurch commented:
@googlebot I am the author of these commits. The build break they caused is also entirely my fault.
